src/sdk/pycli/nnicli/nni_client.py

```diff
@@ -92,48 +72,384 @@ def _create_process(cmd):
+        self.type = None
+        self.sequence = None
+        self.data = None
+        for key in json_obj.keys():
+            self.__setattr__(key, json_obj[key])
+
+    def __repr__(self):
+        return "NNITrialMetricData(timestamp:{} trialJobId:{} parameterId:{} type:{} sequence:{} data:{})" \
+            .format(self.timestamp, self.trialJobId, self.parameterId, self.type, self.sequence, self.data)
+
+class NNITrialHyperParameters:
+    """
+    NNITrialHyperParameters stores the hyper parameters of a trial job.
+
+    Parameters
+    ----------
+    json_obj:
+        json object that stores the hyper parameters
+
+    Attributes
+    ----------
+    parameter_id:
+        parameter id
+    parameter_source:
+        parameter source
+    parameters:
+        hyper parameters
+    parameter_index:
+        parameter index
+    """
+    def __init__(self, json_obj):
+        self.parameter_id = None
+        self.parameter_source = None
+        self.parameters = None
+        self.parameter_index = None
+        for key in json_obj.keys():
+            self.__setattr__(key, json_obj[key])
+
+    def __repr__(self):
+        return "NNITrialHyperParameters(parameter_id:{} parameter_source:{} parameters:{} parameter_index:{})" \
+            .format(self.parameter_id, self.parameter_source, self.parameters, self.parameter_index)
+
+class NNITrialJob:
+    """
+    NNITrialJob stores the information of a trial job.
+
+    Parameters
+    ----------
+    json_obj:
+        json object that stores the hyper parameters
+
+    Attributes
+    ----------
+    id:
+        trial id
+    status:
+        job status
+    hyperParameters:
+        see `NNITrialHyperParameters`
+    logPath:
+        log path
+    startTime:
+        job start time (timestamp)
+    endTime:
+        job end time (timestamp)
+    finalMetricData:
+        see `NNITrialMetricData`
+    parameter_index:
+        parameter index
+    """
+    def __init__(self, json_obj):
+        self.id = None
+        self.status = None
+        self.hyperParameters = None
+        self.logPath = None
+        self.startTime = None
+        self.endTime = None
+        self.finalMetricData = None
+        self.stderrPath = None
+        for key in json_obj.keys():
+            self.__setattr__(key, json_obj[key])
+        if self.hyperParameters:
+            self.hyperParameters = [NNITrialHyperParameters(json.loads(e)) for e in self.hyperParameters]
+        if self.finalMetricData:
+            self.finalMetricData = [NNITrialMetricData(e) for e in self.finalMetricData]
+
+    def __repr__(self):
+        return "NNITrialJob(id:{} status:{} hyperParameters:{} logPath:{} startTime:{} endTime:{} finalMetricData:{} stderrPath:{})" \
+            .format(self.id, self.status, self.hyperParameters, self.logPath,
+                    self.startTime, self.endTime, self.finalMetricData, self.stderrPath)
+
+class NNIExperiment:
+    def __init__(self):
+        self.endpoint = None
```

Comment:
endpoint, port, exp_id should be properties that does not allow external modification.

Reply:
fixed.